src/popup/index.tsx

```diff
@@ -53,10 +56,19 @@ export const Popup = factory(function({
 	const bottomOfVisibleScreen =
 		document.documentElement.scrollTop + document.documentElement.clientHeight;
 	const topOfVisibleScreen = document.documentElement.scrollTop;
+	const widthOfScreen = document.documentElement.clientWidth;
 
 	const willFit = {
 		below: yTop + wrapperDimensions.size.height <= bottomOfVisibleScreen,
-		above: yBottom - wrapperDimensions.size.height >= topOfVisibleScreen
+		above: yBottom - wrapperDimensions.size.height >= topOfVisibleScreen,
+		left:
+			leftStart - wrapperDimensions.size.width >= 0 &&
+			yTop + wrapperDimensions.size.height / 2 <= bottomOfVisibleScreen &&
+			yBottom - wrapperDimensions.size.height / 2 >= topOfVisibleScreen,
```

Comment:
Should this be 
```
(yTop + yBottom)/2 + wrapperDimensions.size.height / 2 <= bottomOfVisibleScreen &&
(yTop + yBottom)/2 - wrapperDimensions.size.height / 2 >= topOfVisibleScreen,
```

Right now I think it's calculating it as if the popup is cut in half with half on top of the trigger and half below it.

Reply:
Yea that's how I have it, so it looks like this
<img width="738" alt="Screen Shot 2020-07-14 at 1 51 37 PM" src="https://user-images.githubusercontent.com/3914018/87475878-82c69a80-c5f3-11ea-9da5-d40958980a6a.png">

I was following this design from the framework that was referenced in the original issue https://ant.design/components/popconfirm/. There was an idea of `bottom-right` and top-right` being added as well.

Reply:
I think that in the most case, top / bottom will be used and only left / right would be used for things like sub-menus or tooltips etc which I believe should be centred.

Reply:
Something to consider is that when placing left / right i think you only need to check the space to the left or right, not the vertical placement. When doing the actual placement though, it should be centred BUT should be bound by the top of the screen.

Reply:
When placing left and right there is a chance of the placement going off screen, which is why I was checking the vertical here, why do you think it doesn't need to be checked?

Reply:
Also what do you mean by being bound by the top of the screen?

Reply:
if you had, for example, a large sub menu coming off of the right side of another menu item you wouldn't want it rendered off of the top of the page, you'd fix it's upper bound to that of the visible window.